src/body.adoc

```diff
@@ -67,3 +67,636 @@ The Ssplcofi extension guarantees that, for Zihpm counters programmed to count p
+
+The machine-level extension, Smpdis, encompasses all newly added CSRs and behavior modifications for a hart across all privilege levels. The corresponding supervisor-level extension, Sspdis, is essentially identical to Sspdis, except that it excludes machine-level CSRs and behaviors not intended to be directly accessible at the supervisor level.
+
+NOTE: _When to use event-based sampling (like Sspesa or Ssplcofi) vs instruction-based sampling (Smpids/Sspdis) is an art rather than a science.  Say more here._
+
+=== CSRs
+
+[[pdisctl]]
+==== Machine PDIS Control Register (`mpdisctl`)
+
+The `mpdisctl` register is a 64-bit WARL register that configures the PDIS mechanism.  For RV32, bits 63:32 of `mpdisctl` can be accessed via the `mpdisctlh` CSR.
+
+WARNING: _do we need RV32 support for PDIS?  Are such processors likely to implement PDIS?  We skipped it for CTR..._
+
+.Machine PDIS Control Register (`mpdisctl`)
+[wavedrom, , svg]
+....
+{reg: [
+    {bits:  3, name: 'SEL'},
+    {bits: 29, name: 'HPM[31:3]'},
+    {bits:  1, name: 'MEM'},
+    {bits:  1, name: 'ACC'},
+    {bits:  1, name: 'TS'},
+    {bits:  1, name: 'PA'},
+    {bits:  1, name: 'PTRG'},
+    {bits: 20, name: 'WARL', type: 1},
+    {bits:  3, name: 'Custom'},
+    {bits:  1, name: 'U'},
+    {bits:  1, name: 'S'},
+    {bits:  1, name: 'M'},
+    {bits:  1, name: 'OF'},
+], config:{lanes: 4, hspace:1024}}
+....
+
+[cols="15%,85%",options="header"]
+|====
+| Field | Description
+| SEL | Selects the types of instructions to be counted and sampled.  Encodings are as follows.
+
+0: All instructions decoded.
+
+1: Load instructions decoded.
+
+2: Store instructions decoded.
+
+3: Load and store instructions decoded.
+
+4: Control transfer instructions decoded.
+
+5..7: _Reserved_
+
+| HPM[31:3] | Each HPM__i__ bit enables recording of whether the sampled instruction incurred the event selected in `mhpmevent__i__`.  Recording of events for HPM__i__ requires that counter __i__ is delegated to the same privilege mode as PDIS, that the selected event supports PDIS recording, and that the counter is enabled in the privilege mode in which the sampled instruction executes.
+| MEM | When set, enables storing of sample records (<<samprec>>) to the PDIS Memory Buffer (<<membuff>>).  When cleared, sample records are instead written to `spdisdata__i__` registers (<<dataregs>>).
+| ACC | Accelerate instruction selection after a sample is filtered out or otherwise discarded.  See <<sampsel>>.
+| TS | Enable recording of a timestamp per record.  See <<samprec>>.
+| PA | Enable recording the physical address of sampled explicit memory operations.  See <<samprec>>.
+| PTRG | Enable recording of the prior control transfer target.  See <<samprec>>.
+| Custom | WARL bits designated for custom use.  The value 0 must correspond to standard behavior.  See <<Custom Extensions>>.
```

Comment:
Would it be better to reserve these at the end? Also note in the description how many bits are available.